script.js

```diff
@@ -0,0 +1,185 @@
+// api url
+const API_URL =
+    "https://raw.githubusercontent.com/alexsimkovich/patronage/main/api/data.json";
+// declaration of items creating product list
+const ulList = document.querySelector(".product__container");
+const loadingSpinner = document.querySelector(".spinner");
+// declaration of items creating cart
+const ulListCart = document.querySelector(".cart__list");
+const cartEmpty = document.querySelector(".cart__empty");
+const cartTotal = document.querySelector(".cart__total");
+
+// The array which store all items from API
+let pizzaList = [];
+// The array which store all items in cart
+let cartArr = [];
+
+const fetchData = async () => {
+    const response = await fetch(API_URL);
+    if (!response.ok) {
+        const message = `An error has occured: ${response.status}`;
+        throw new Error(message);
+    }
+    return response.json();
+};
+
+const renderData = async () => {
+    try {
+        await fetchData().then((data) => {
+            loadingSpinner.classList.add("spinner--hide");
+            renderPizzaList(data);
+        });
+    } catch (error) {
+        loadingSpinner.classList.add("spinner--hide");
+        renderError(error);
+    }
+};
+
+renderData();
+
+const renderError = (error) => {
+    const errorInfo = document.createElement("p");
+    errorInfo.classList.add("product__item", "product__item--error");
+    errorInfo.innerText = `${error.message}`;
+    ulList.appendChild(errorInfo);
+};
+
+const renderPizzaList = (data) => {
+    pizzaList = data;
+
+    data.map((item) => {
+        const listItem = document.createElement("li");
+        listItem.classList.add("product__item");
+        listItem.setAttribute("id", `${item.id}`);
+        ulList.appendChild(listItem);
+
+        listItem.innerHTML = `
+		<img
+			src="${item.image}"
+			alt="${item.title}"
+			class="product__img"
+		/>
+		<div class="product__box">
+			<h3>${item.title}</h3>
+			<p class="product__description">
+			${item.ingredients.join(", ")}
+			</p>
+		</div>
+		<div class="product__actions">
+			<p class="product__price">${item.price.toFixed(2)}</p>
+			<button type="button" class="button button--order">Zamów</button>
+		</div>
+		`;
+    });
+
+    const orderButtons = ulList.querySelectorAll(".button--order");
+    orderButtons.forEach((btn) => {
+        btn.addEventListener("click", orderButtonHandler);
+    });
+};
+
+// the function executes when order button is clicked
+const orderButtonHandler = (e) => {
+    if (e.target.matches(".button--order")) {
```

Comment:
po co ten if?

Reply:
Ten if był potrzebny, gdy miałem nasłuchwianie całą ul listę, a nie na same buttony. Nie zauważyłem, żeby to usunąć jak zmieniałem listenery. Poprawiłem do samo w funkcji removeButtonHandler.